src/order/prime_ideal.lean

```diff
@@ -163,6 +163,42 @@ end
 lemma is_prime.mem_compl_of_not_mem (hI : is_prime I) (hxnI : x ∉ I) : xᶜ ∈ I :=
 hI.mem_or_compl_mem.resolve_left hxnI
 
+lemma is_prime_of_mem_or_compl_mem [is_proper I] (h : ∀ {x : P}, x ∈ I ∨ xᶜ ∈ I) : is_prime I :=
+begin
+  rw is_prime_iff_mem_or_mem,
+  intros x y hxy,
+  rw or_iff_not_imp_left,
+  intro hxI,
+  have hxcI : xᶜ ∈ I,
+  exact h.resolve_left hxI,
```

Comment:
It looks like there were multiple goals open. Please use braces `{ .. }` to structure the proof. See also the third example in the `Tactic mode` section of the style guide: https://leanprover-community.github.io/contribute/style.html#tactic-mode